lib/valkyrie/persistence/solr/query_service.rb

```diff
@@ -55,6 +55,13 @@ def find_all_of_model(model:)
       Valkyrie::Persistence::Solr::Queries::FindAllQuery.new(connection: connection, resource_factory: resource_factory, model: model).run
     end
 
+    # Count all of the Valkyrie Resources of a model persisted in the Solr index
+    # @param [Class, String] model the Valkyrie::Resource Class
+    # @return integer
+    def count_all_of_model(model:)
+      Valkyrie::Persistence::Solr::Queries::FindAllQuery.new(connection: connection, resource_factory: resource_factory, model: model).run.count
```

Comment:
We should be able to do a query and just get the count from the Solr response, instead of retrieving each resource just to count them.

Reply:
@escowles would adding a `count` method to `FindAllQuery` that doesn't require the use of `run` and `each` make sense to achieve this?

Reply:
@dgcliff Yes, I think the `FindAllQuery` with the model param supplied is probably a good base for this.  Then you'd need to get the total number of records from RSolr instead of instantiating the resources to count them.